packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/hooks/usePaginationState.ts

```diff
@@ -0,0 +1,12 @@
+import { parseAsJson, useQueryState } from "nuqs";
+import { PaginationState } from "@tanstack/react-table";
+
+export function usePaginationQueryState(defaultPageSize = 10) {
+  return useQueryState(
+    "pagination",
+    parseAsJson<PaginationState>().withDefault({
+      pageIndex: 0,
+      pageSize: defaultPageSize,
+    }),
+  );
```

Comment:
would this make sense to get nicer query params?
```suggestion
  const [page, setPage] = useQueryState("page", parseAsNumber().withDefault(0));
  const [pageSize, setPageSize] = useQueryState("pageSize", parseAsNumber().withDefault(defaultPageSize));
  return { page, setPage, pageSize, setPageSize };
```

Reply:
Changed it slightly so it returns a tuple like `useQueryState` but yeah, this works nicely 👌